src/s390x/zipl.rs

```diff
@@ -33,11 +34,116 @@ pub fn chreipl<P: AsRef<Path>>(dev: P) -> Result<()> {
     Ok(())
 }
 
+/// Secure boot (Secure IPL) includes support of SCSI and ECKD DASD boot devices
+enum Loaddev {
+    Eckd(String),
+    Scsi(String, String, String),
+}
+
+fn parse_lszdev_eckd(line: &str) -> Result<Loaddev> {
+    // ECKD ID looks like: 0.0.5223, we need only last part of it (5223)
+    lazy_static! {
+        static ref REGEX: Regex = Regex::new(r#"[[:digit:]].[[:digit:]].([[:xdigit:]]+)"#).unwrap();
+    }
+    if let Some(cap) = REGEX.captures_iter(line).next() {
+        return Ok(Loaddev::Eckd(cap[1].to_string()));
+    }
+    bail!("bad ECKD id: {}", line);
+}
+
+fn parse_lszdev_zfcp(line: &str) -> Result<Loaddev> {
+    // SCSI ID looks like: 0.0.8000:0x500507630400d1e3:0x4000401d00000000
+    // So here is regex to parse required ids: 8000,500507630400d1e3,4000401d00000000
+    lazy_static! {
+        static ref REGEX: Regex = Regex::new(
+            r#"[[:digit:]].[[:digit:]].([[:xdigit:]]+):0x([[:xdigit:]]+):0x([[:xdigit:]]+)"#
+        )
+        .unwrap();
+    }
+    if let Some(cap) = REGEX.captures_iter(line).next() {
+        return Ok(Loaddev::Scsi(
+            cap[1].to_string(),
+            cap[2].to_string(),
+            cap[3].to_string(),
+        ));
+    }
+    bail!("bad zFCP id: {}", line);
+}
+
+fn parse_lszdev<P: AsRef<Path>>(dev: P) -> Result<Loaddev> {
+    // We don't want to traverse sysfs and do same stuff lszdev does,
+    // so just call it to get required info. Here is sample output:
+    // $ lszdev -c TYPE,ID -n
```

Comment:
I assume you checked with `--list-columns` that there aren't more targeted columns we could use to avoid regexing things?